### DISCUSSION TITLE
RememberMe behavior for SocialMedia/External User Accounts

### DISCUSSION THREAD

**Discussion by ddurham2** [keycloak/keycloak#16182] [Q&A]
I've noticed that when a local user account authenticates with Remember Me checked, then it changes the lifetime of the KEYCLOAK_IDENTITY cookie from session to the remember-me SSO lifetime.  This makes sense.

However, when a user authenticates against google/facebook/etc the KEYCLOAK_IDENTITY cookie is always session lifetime, and the resulting undesirable behavior is that the user must authenticate again after a browser restart (or in my case, also when a mobile device is restarted or when the mobile app is killed and restarted).

I have searched high and low for a setting that enables remember-me behavior for a configured OIDC/social media IDPs, but to no avail.

[Looking at the code](https://github.com/keycloak/keycloak/blob/de7dd77aebe5f81cf108d4e642400feddcbae150/services/src/main/java/org/keycloak/services/managers/AuthenticationManager.java#L783), the cookie's lifetime is determined by the isRememberMe() property of the UserSession object.

On a hunch, I tried defining a mapper under the Identity Provider configuration that sets a user-session property of rememberMe to true, but it only seems to let you define string type properties.  And no other values like "1" that I tried seemed to give the desired change in cookie expiration either.

So, I'm curious if anyone has ever gotten this working, or maybe I'm missing something.  It seems like a perfectly reasonable thing that a user not have to re-login repeatedly even if they use an external IDP to do so.

I'm unsure if there's a way, from server configuration, to manipulate the user session's values before that cookie is set.  I did dig around in the setup at google and facebook to try and return a property in the token (of the boolean = true type) that I could try and define a mapper for, but I didn't see any such setting.

Please advise as to how I can configure things to work.  Thanks!

**Comment by zemadz** [comment]
This should be achievable by a custom authenticator. An example of this is here https://github.com/Herdo/keycloak-remember-me-authenticator/tree/main.

The added extension should be configured for the login flow that you are using. This discussion was helpful for me https://keycloak.discourse.group/t/guidance-for-enabling-remember-me-functionality-for-social-providers/7188/2.

**Comment by ddurham2** [comment]
Thanks @zemadz, https://github.com/Herdo/keycloak-remember-me-authenticator/tree/main worked great out of the box.

Really wish control over this were built into keycloak/

**Comment by mandree95** [comment]
I would suggest this should / could be an option flag in the identity provider settings. Is this on the roadmap?

**Comment by yosefy** [comment]
yes, we need to rebuild the image each time just to add this functionality
it is not secure and also much lots  of work.
what is the expected proper way to achieve this?
thanks
